src/apps/material/helper.ts

```diff
@@ -454,6 +457,35 @@ export const useGetHoldings = ({
   return { data, isLoading, isError };
 };
 
+export const getManifestationBasedOnType = (
+  work: Work,
+  materialType: DisplayMaterialType
+): Manifestation => {
+  const { bestRepresentation, all } = work.manifestations;
+
+  // Retrieves the material type of the best representation.
+  const bestRepresentationMaterialType =
+    getManifestationMaterialTypes(bestRepresentation);
+
+  // Checks if the best representation matches the desired material type.
+  if (materialType === bestRepresentationMaterialType) {
+    return bestRepresentation;
+  }
+
+  // Filters and sorts the manifestations if the best representation does not match.
+  const filteredByType = filterManifestationsByType(materialType, all);
+  const sortedByYear = orderManifestationsByYear(filteredByType);
+
+  // Returns the newest manifestation based on type if available.
+  const newestManifestationBasedOnType = sortedByYear[0];
+  if (newestManifestationBasedOnType) {
```

Comment:
Consider using `first()` from lodash instead of `[0]`.

[I have described why in Zulip](https://reload.zulipchat.com/#narrow/stream/240325-DDF/topic/.5B0.5D.20.3D.3E.20_.2Ehead.28.29/near/339889428) but I see that you have implemented a check which is good.